src/dragonball/src/tracer.rs

```diff
@@ -0,0 +1,176 @@
+// Copyright (C) 2023 Alibaba Cloud. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::sync::Arc;
+
+use anyhow::Result;
+use derivative::Derivative;
+use log::{info, warn};
+use tracing::subscriber::DefaultGuard;
+use tracing::{span::EnteredSpan, subscriber::NoSubscriber, trace_span, Subscriber};
+
+/// Error for dragonball tracer.
+#[derive(Debug, thiserror::Error)]
+pub enum TraceError {
+    /// Tracing is already enabled.
+    #[error("Tracing is already enabled.")]
+    AlreadyEnabled,
+}
+
+/// Configuration information for tracing feature.
+#[derive(Clone, Derivative)]
+#[derivative(Debug, PartialEq, Eq)]
```

Comment:
Why `PartialEq` and `Eq` are needed?
Not seeing comparasion between different `TraceInfo` in your code.

Reply:
`TraceInfo` is used in `VmmAction::SetHypervisorTracing(TraceInfo)` and `VmmAction` implements the `PartialEq` and `Eq`. This is also the reason why I added the `sid` field to `TraceInfo`.